packages/ERTP/src/eventualWithdraw.js

```diff
@@ -0,0 +1,40 @@
+// @ts-check
+import { E } from '@agoric/eventual-send';
+import { makePromiseKit } from '@agoric/promise-kit';
+
+import './types';
+
+/** @type {EventualWithdraw} */
+export const eventualWithdraw = (
+  purse,
+  amountMath,
+  searchAmount,
+  iterations = 1,
```

Comment:
I like where this is going.

My only comment is that I would prefer additionally returning something like a cancellation token so that the caller can abort withdrawal in a race-free way (the returned `Promise<Payment>` may still be resolved if the cancellation happens after the withdrawal).

I'm not sure of the best mechanism, but counting the number of iterations is rather arbitrary and may not correspond to anything useful.  Other suggestions are welcome, too.